airflow/providers/common/sql/sensors/sql.py

```diff
@@ -96,19 +96,37 @@ def poke(self, context: Any):
         records = hook.get_records(self.sql, self.parameters)
         if not records:
             if self.fail_on_empty:
-                raise AirflowException("No rows returned, raising as per fail_on_empty flag")
+                # TODO: remove this if block when min_airflow_version is set to higher than 2.7.1
+                message = "No rows returned, raising as per fail_on_empty flag"
+                if self.soft_fail:
+                    raise AirflowSkipException(message)
+                raise AirflowException(message)
```

Comment:
I think it's cleaner to have it like:

https://github.com/apache/airflow/blob/47bd5dd0e1e13af45206b94dd5518ada278a9552/airflow/providers/amazon/aws/log/s3_task_handler.py#L36-L42

Reply:
This is for avoiding breaking changes as suggested in https://github.com/apache/airflow/pull/33405#discussion_r1306449016